prow/crier/reporters/gerrit/reporter.go

```diff
@@ -432,17 +456,55 @@ func GenerateReport(pjs []*v1.ProwJob, commentSizeLimit int) JobReport {
 		return true
 	})
 
-	// Truncate if there is not enough space left
 	remainingSize := commentSizeLimit - len(report.Header)
+	linesWithURLs := make([]string, len(report.Jobs))
+	linesWithoutURLs := make([]string, len(report.Jobs))
 	for i, job := range report.Jobs {
-		message := job.serialize() + "\n"
-		remainingSize -= len(message)
-		if remainingSize < 0 {
-			report.Message += fmt.Sprintf("[Skipped %d/%d jobs due to reaching gerrit comment size limit]\n", len(report.Jobs)-i, len(report.Jobs))
-			break
+		linesWithURLs[i] = job.serialize()
+		remainingSize -= len(linesWithURLs[i])
+	}
+
+	// cutoff is the index where if it's the line contains URL it will exceed the
+	// size limit.
+	cutoff := len(report.Jobs)
+	for cutoff > 0 && remainingSize < 0 {
+		previousShortLine := report.Jobs[cutoff-1].serializeWithoutURL()
+		linesWithoutURLs[cutoff-1] = previousShortLine
+		// remainingSize >= 0 after this condition means next line is not good
+		// to include URL
+		remainingSize += len(linesWithURLs[cutoff-1]) - len(linesWithoutURLs[cutoff-1])
+		cutoff--
+	}
+
+	// Should recommend /test all when URL is skipped on failed/aborted prowjobs
+	if cutoff != len(report.Jobs) &&
+		(report.Jobs[cutoff].State == v1.FailureState || report.Jobs[cutoff].State == v1.AbortedState) {
+		reTestMessage = " Comment '/test all' to rerun all failed tests"
+	}
+
+	// Now there is enough info, we can finalize Header
+	report.Header += fmt.Sprintf("%s\n", reTestMessage)
+
+	// This shouldn't happen unless there are too many prow jobs(e.g. > 300) and
+	// each job name is super long(e.g. > 50)
+	if remainingSize < 0 {
+		report.Message = "Prow failed to report all jobs, are there excessive amount of prow jobs?"
```

Comment:
Crier doesn't have this info yet, will need some more refactoring to add this nicely. Since the previous `triggered N jobs` message already presented the link, I think it's fine to omit here